app/scripts/modules/core/src/search/widgets/Search.tsx

```diff
@@ -267,13 +296,18 @@ export class Search extends React.Component<ISearchProps, ISearchState> {
     }
   }
 
+  private handleMouseDown(): void {
+    this.mouseDownFired = true;
+  }
+
   private handleClearClick(): void {
     this.setState({
       isOpen: false,
       tags: []
     });
     this.inputElement.value = '';
```

Comment:
Did you try to actually bind `inputElement`s `value` to state and then modify the state `onChange` instead of stepping outside of React for this? (https://facebook.github.io/react/docs/forms.html)

Reply:
Stop reviewing old code, mister....
I didn't, because I was completely unaware about that - I'll flag that for investigation.

Reply:
I reviewed this at 10am... it was fresh then!